app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
-          && service.adapter !== 'ip-camera@link.mozilla.org'
-      ))
-      // Keep only the services with getters.
-      .then(services => services.filter(service =>
-        Object.keys(service.getters).length
-      ))
-      // User friendly name.
-      .then(services => services.map(service => {
-          switch (service.adapter) {
-            case 'clock@link.mozilla.org':
-              service.name = 'Everyday';
-              break;
-
-            default:
-              service.name = service.adapter;
-              break;
+    // Fetch recipe enabled statuses and sources.
+    return recipeServicesPromise.then(services => {
+      // Fetch their respective enabled status.
+      const { enabledSelectors, sourceSelectors } = services.reduce(
+        (selectors, service) => {
+
+          selectors.enabledSelectors.push({ id: service.getEnabled });
+          selectors.sourceSelectors.push({ id: service.getSource });
+
+          return selectors;
+        },
+        { enabledSelectors: [], sourceSelectors: [] }
+      );
+
+      const getterURL = `${this[p.net].origin}/api/` +
+        `v${this[p.settings].apiVersion}/channels/get`;
+
+      return Promise.all([
+        this[p.net].fetchJSON(getterURL, 'PUT', enabledSelectors),
+        this[p.net].fetchJSON(getterURL, 'PUT', sourceSelectors)
+      ])
+      .then(([statuses, sources]) => {
+        return services.map(service => {
+          const statusResponse = statuses[service.getEnabled];
+          const sourceResponse = sources[service.getSource];
+
+          if (statusResponse && statusResponse.OnOff) {
+            service.status = statusResponse.OnOff === 'On';
+          } else {
+            console.error(
+              'Error occurred while retrieving recipe (%s) status: ',
+              service.id,
+              statusResponse && statusResponse.Error
+            );
+            service.status = false;
           }
 
-          return service;
-        }
-      ))
-      // User friendly getters.
-      .then(services => services.map(service => {
-          switch (service.adapter) {
-            case 'clock@link.mozilla.org':
-              service.getters = [
-                {
-                  label: 'in the morning',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '08:00am'
-                },
-                {
-                  label: 'in the afternoon',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '02:00pm'
-                },
-                {
-                  label: 'in the evening',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '06:00pm'
-                }
-              ];
-              break;
-
-            default:
-              //service.getters = service.getters;
-              break;
+          if (sourceResponse && sourceResponse.String) {
```

Comment:
Just basic guards for the cases I've mentioned above, once we have stable API we can more intelligent and generic checks.